shared/signup/phone-number/phone-input.tsx

```diff
@@ -1,36 +1,116 @@
 import * as React from 'react'
 import * as Kb from '../../common-adapters'
 import * as Styles from '../../styles'
-import {isIOS} from '../../constants/platform'
-import {countryData, AsYouTypeFormatter, validateNumber} from '../../util/phone-numbers/'
+import {isIOS, isMobile} from '../../constants/platform'
+import {
+  countryData,
+  codeToCountry,
+  areaCodeIsCanadian,
+  AsYouTypeFormatter,
+  validateNumber,
+} from '../../util/phone-numbers'
 import {memoize} from '../../util/memoize'
 
-const getCallingCode = countryCode => countryData[countryCode].callingCode
-const getCountryEmoji = countryCode => <Kb.Emoji size={16} emojiName={countryData[countryCode].emojiText} />
-const getPlaceholder = countryCode => 'Ex: ' + countryData[countryCode].example
+const normalizeCode = countryCode => (countryCode.endsWith('?') ? countryCode.slice(0, -1) : countryCode)
+const getCallingCode = countryCode =>
+  countryCode !== '' ? countryData[normalizeCode(countryCode)].callingCode : ''
+const getCountryEmoji = countryCode => (
+  <Kb.Emoji size={16} emojiName={countryData[normalizeCode(countryCode)].emojiText} />
+)
+const getPlaceholder = countryCode =>
+  countryCode !== '' ? 'Ex: ' + countryData[normalizeCode(countryCode)].example : 'N/A'
 const filterNumeric = text => text.replace(/[^0-9]/g, '')
 const defaultCountry = 'US'
+const prioritizedCountries = ['US', 'CA', 'GB']
+
 const pickerItems = memoize(countryData =>
-  Object.values(countryData)
-    .sort((a: any, b: any) => a.name.localeCompare(b.name))
-    .map((cd: any) => ({label: cd.pickerText, value: cd.alpha2}))
+  [
+    ...prioritizedCountries.map(code => countryData[code]),
+    ...Object.values(countryData)
+      .sort((a: any, b: any) => a.name.localeCompare(b.name))
+      .map((cd: any) => {
+        if (prioritizedCountries.includes(cd.alpha2)) {
+          return {
+            ...cd,
+            alpha2: cd.alpha2 + '?',
+          }
+        }
+        return cd
+      }),
+  ].map((cd: any) => ({label: cd.pickerText, value: cd.alpha2}))
 )
-const menuItems = memoize((countryData, filter, onClick) =>
-  Object.values(countryData)
-    .sort((a: any, b: any) => a.name.localeCompare(b.name))
+const menuItems = memoize((countryData, filter, onClick) => {
+  const strippedFilter = filter.replace(/[^\d]/g, '')
+  const lowercaseFilter = filter.toLowerCase()
+
+  return Object.values(countryData)
     .filter((cd: any) => {
```

Comment:
I think this code will be written once and then never reworked again so I doubt types would do anything for us now. If we ever reuse the phone number utils module somewhere else, absolutely.

Reply:
I think it'd be nice to add the type regardless, it's just one object and it will be easier to do it now while you're in here than for someone else down the line.

Reply:
k done

Reply:
thanks!